vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -9,15 +9,58 @@
 #include "cb_adf.h"
 #include "cb_algs.h"
 #include "constant.h"
+#include "object_pool.h"
 
 #include <numeric>
 #include <algorithm>
 #include <unordered_set>
+#include <queue>
 
 using namespace LEARNER;
 using namespace VW;
 using namespace VW::config;
 
+template <typename T>
+struct v_array_initializer
+{
+  v_array<T>* operator()(v_array<T>* array)
+  {
+    *array = v_array<T>();
```

Comment:
This seems ... confusing?  You pass in a pointer and then you assign to it, and then you return it.  Why assign?  And why pass in anything?  Interface-wise it looks like a memory leak trap.

Reply:
The object works on pre-allocated memory. This is an initializer for the object, and so the convention would be to do a placement new into the memory. The difficultly is that technically you need to use the return value of the placement new to reference the object not the original pointer. This is why there is a return value.

The expected behavior is to modify the pointer passed in to initialize it and then return it.